windows/main.cpp

```diff
@@ -128,3 +128,12 @@ void uiQueueMain(void (*f)(void *data), void *data)
 		// LONGTERM this is likely not safe to call across threads (allocates memory)
 		logLastError(L"error queueing function to run on main thread");
 }
+
+void uiTimer(int milliseconds, int (*f)(void *data), void *data)
+{
+	UINT_PTR timer;
+	
+	timer = (UINT_PTR) new TimerHandler(f, data);
```

Comment:
Are you sure you can use a pointer to an arbitrary structure here? 
From what I understood of the win32 docs, if you give it both an hWnd and a nIDEvent arguments, the function expect it to point to an existing timer id linked to that window.

Reply:
> [Timer IDs are pointer-sized](https://blogs.msdn.microsoft.com/oldnewthing/20150924-00/?p=91521), so we can get away with using the pointer to the timer metadata struct itself as an ID; should probably consider doing that instead.

Reply:
Sorry I missed that comment... that's really an interesting design.